### PR TITLE
Add StringLower operator

### PR DESCRIPTION
**Pull request by xadupre**
StringLower works on any string on Linux, MacOs. On Windows, every non ascii string still fails. onnxruntime may have to be compiled with option utf-8. Still under investigation.

### CODE REVIEW

**Comment by wenbingl** [comment]
Need fix the test before I approve the PR.

**Comment by xadupre** [comment]
> Need fix the test before I approve the PR.

The test is fixed.

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,6 +87,9 @@ if (OCOS_ENABLE_SPM_TOKENIZER)
   list(APPEND TARGET_SRC ${stpiece_TARGET_SRC})
 endif()
 
+
+add_compile_options("$<$<C_COMPILER_ID:MSVC>:/utf-8>")
```
Comment (wenbingl) [inline CMakeLists.txt:91]: Did you try to save the UTF-8 with BOM? MSVC should be OK with it then.

**Review by wenbingl — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (xadupre) [reply CMakeLists.txt:91]: The BOM was added to the file including chinese. The flag utf-8 is used to change the way strings are processed: https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-160. I may try other scenario if you prefer.

**Pull request merged**